### PR TITLE
fix mmseg twice resize

### PR DESCRIPTION
**Pull request by RunningLeon**
## Motivation

Unnecessary resize in mmseg encode-decoder head: https://github.com/open-mmlab/mmsegmentation/issues/1529

## Modification

- [x] Remove second unused resize.
- [x] Make sure output of 4d tensor in a nice way.
- [x] Update unit test.

## BC-breaking (Optional)

None

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

### CODE REVIEW

File: mmdeploy/codebase/mmseg/models/segmentors/encoder_decoder.py [hunk 1]
```diff
@@ -25,16 +23,7 @@ def encoder_decoder__simple_test(ctx, self, img, img_meta, **kwargs):
...
-        mode='bilinear',
-        align_corners=self.align_corners)
     seg_logit = F.softmax(seg_logit, dim=1)
-    seg_pred = seg_logit.argmax(dim=1)
-    # our inference backend only support 4D output
-    shape = seg_pred.shape
-    if not is_dynamic_shape(ctx.cfg):
-        shape = [int(_) for _ in shape]
-    seg_pred = seg_pred.view(shape[0], 1, shape[1], shape[2])
+    # keep 4D output required by backends such as TensorRT
```
Comment (grimoire) [inline mmdeploy/codebase/mmseg/models/segmentors/encoder_decoder.py:27]: It is not required by any backends.
We do a nearest resize outside the model in `segmentation_model` as the postprocess, which require 4d input.

**Review by grimoire — COMMENTED** [review]
Moving the `resize` out of the model is a good optimization.
But the accuracy of the model might be affected. should we update the benchmark?

File: mmdeploy/codebase/mmseg/models/segmentors/encoder_decoder.py (hunk 1 above)
Comment (RunningLeon) [reply mmdeploy/codebase/mmseg/models/segmentors/encoder_decoder.py:27]: It seems that trt requires 4d output as tested before when supporting deployment feature in mmseg

Comment (grimoire) [reply mmdeploy/codebase/mmseg/models/segmentors/encoder_decoder.py:27]: It is caused by postprocessing even in mmseg. We can just remove this comment.

**Review by grimoire — APPROVED** [review]

**Review by AllentDan — APPROVED** [review]
LGTM
